core/templates/pages/contributor-dashboard-page/contributor-stats/contributor-stats.component.html

```diff
@@ -0,0 +1,601 @@
+                  class="dashboard-table-headings stats-data">
+                  <div class="stats-column-name-container">
+                      {{column.name}}
+                  </div>
+              </th>
+            </tr>
+            <tr *ngFor="let row of block.pagedStats[block.currentPage - 1].stats" class="stats-row">
+              <td class="stats-data">
+                  {{row.firstContributionDate}} - {{row.lastContributionDate}}
+              </td>
+              <td class="stats-data">
+                  {{row.topicName}}
+              </td>
+              <td *ngIf="type == 'translationReview'" class="stats-data">
+                  {{row.reviewedCards}}
+              </td>
+              <td *ngIf="type == 'translationReview'" class="stats-data">
+                  {{row.reviewedWordCount}}
+              </td>
+              <td *ngIf="type == 'translationContribution' || type == 'translationReview'" class="stats-data">
+                  {{row.acceptedCards}}
+              </td>
+              <td *ngIf="type == 'translationContribution' || type == 'translationReview'" class="stats-data">
+                  {{row.acceptedWordCount}}
+              </td>
+            </tr>
+          </table>
+          <div *ngIf="type == 'translationContribution'" class="mobile-stats-block">
+            <div class="mobile-stats-data-container" *ngFor="let row of block.pagedStats[block.currentPage - 1].stats">
+              <div class="mobile-stats-row">
+                <div class="mobile-stats-name">
+                  <b>{{COLUMNS[type][0].name}}</b>
+                </div>
+                <div>
+                  {{row.firstContributionDate}} - {{row.lastContributionDate}}
+                </div>
+              </div>
+              <div class="mobile-stats-row">
+                <div class="mobile-stats-name">
+                  <b>{{COLUMNS[type][1].name}}</b>
+                </div>
+                <div>
+                  {{row.topicName}}
+                </div>
+              </div>
+              <div class="mobile-stats-row">
+                <div class="mobile-stats-name">
+                  <b>{{COLUMNS[type][2].name}}</b>
+                </div>
+                <div>
+                  {{row.acceptedCards}}
+                </div>
+              </div>
+              <div class="mobile-stats-row">
+                <div class="mobile-stats-name">
+                  <b>{{COLUMNS[type][3].name}}</b>
+                </div>
+                <div>
+                  {{row.acceptedWordCount}}
+                </div>
+              </div>
+            </div>
+          </div>
+          <div *ngIf="type == 'translationReview'" class="mobile-stats-block">
+            <div class="mobile-stats-data-container" *ngFor="let row of block.pagedStats[block.currentPage - 1].stats">
+              <div class="mobile-stats-row">
+                <div class="mobile-stats-name">
+                  <b>{{COLUMNS[type][0].name}}</b>
```

Comment:
I worry about this use of [0], [1], [2], [3] -- I have no idea from reading this what it represents. Can we use dicts instead as the internal representation, rather than lists, so that this is more readable? (You're already going to be listing it out here anyway.)

The benefit of using a list is if you have an ng-repeat here but since you're just repeating blocks of code manually I don't think that representation is particularly useful (unless you can find a way to simplify the display logic and make it generic for each row, e.g. you could compute a new property string for the dates and reference the corresponding property name in a generic for loop -- that would probably be better and make the code simpler and more declarative, but if you can't pull it off then it's better to use a dict instead to avoid the [0], [1], [2], ..., [5]).

Ditto below/elsewhere.